libpod/image/image.go

```diff
@@ -149,43 +210,344 @@ func (i *Image) ID() string {
 	return i.image.ID
 }
 
-// createNamesToPull looks at a decomposed image and determines the possible
-// images names to try pulling in combination with the registries.conf file as well
-func (i *Image) createNamesToPull() ([]string, error) {
-	var pullNames []string
-	decomposedImage, err := decompose(i.InputName)
+// Digest returns the image's Manfiest
+func (i *Image) Digest() digest.Digest {
+	return i.image.Digest
+}
+
+// Names returns a string array of names associated with the image
+func (i *Image) Names() []string {
+	return i.image.Names
+}
+
+// Created returns the time the image was created
+func (i *Image) Created() time.Time {
+	return i.image.Created
+}
+
+// TopLayer returns the top layer id as a string
+func (i *Image) TopLayer() string {
+	return i.image.TopLayer
+}
+
+// Remove an image; container removal for the image must be done
+// outside the context of images
+func (i *Image) Remove(force bool) error {
+	_, err := i.imageruntime.store.DeleteImage(i.ID(), true)
+	return err
+}
+
+func (i *Image) GetImageInspectInfo() (*inspect.ImageData, error) {
+
+	layer, err := i.imageruntime.store.Layer(i.TopLayer())
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading information about layer %q", i.TopLayer())
+	}
+	// TODO: I think this might the wrong way to get this? double check it
+	size, err := i.imageruntime.store.DiffSize(layer.Parent, layer.ID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error determining size of layer %q", layer.ID)
+	}
+	driverData, err := driver.GetDriverData(i.imageruntime.store, layer.ID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting graph driver info %q", i.ID())
+	}
+	return i.getImageData(size, driverData)
+}
+
+func (i *Image) getImageData(size int64, driver *inspect.Data) (*inspect.ImageData, error) {
```

Comment:
Can this be combined with the above GetImageInspectInfo()? I don't see any use to keeping them separate

Reply:
on further thinking, i dont believe either func should be in here ... they should remain in runtime_img because they are only consumed by libpod. Removing.